### ISSUE TITLE
[BUG] Error 'Scaffold is not a function' when executing from Node.js

### ISSUE THREAD

**Issue by opack** [chenasraf/simple-scaffold#67]
#### Describe the bug

Hello! I wanted to use custom helpers to be able to determine the name of my component from the given `name`, which in my case would be a path.
I tried a few things with the config, but then I saw somewhere in the doc that the `helpers` is only accessible in the Node.JS usage. So I tried using it, but always fail with `Scaffold is not a function`. So, I took the example from the doc ([here](https://chenasraf.github.io/simple-scaffold/pages/node.html) and [here](https://chenasraf.github.io/simple-scaffold/pages/templates.html#md:equivalent-node-module-example)), but still get the same error.
Either I'm doing something wrong, or there is a bug. Could you help please?

#### How To Reproduce

1. Here is my file `scaffold.js` (config from [doc](https://chenasraf.github.io/simple-scaffold/pages/node.html)):

   ```javascript
   import Scaffold from 'simple-scaffold'

   const config = {
	name: 'component',
	templates: ['tools/templates/page'],
	output: '.',
	createSubFolder: true,
	subFolderNameHelper: 'upperCase',
	data: {
	   property: 'value'
	},
	helpers: {
	   twice: (text) => [text, text].join(' ')
	},
	beforeWrite: (content, rawContent, outputPath) => content.toString().toUpperCase()
   }

   const scaffold = Scaffold(config)
   ```
Similar result (same error) when I use this config from [this doc](https://chenasraf.github.io/simple-scaffold/pages/templates.html#md:equivalent-node-module-example):

2. Run with args/config:

   ```shell
   node scaffold.js
   ```

#### Expected behavior\*\*

Pages are generated. At least, no error.

#### Logs

``` shell
$ node scaffold.js
file:///C:/Users/micro/Projects/kazelio-kronos/scaffold.js:18
const scaffold = Scaffold(config)
                 ^

TypeError: Scaffold is not a function
    at file:///C:/Users/micro/Projects/kazelio-kronos/scaffold.js:18:18
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)

Node.js v18.17.1
```

#### Desktop (please complete the following information):

- OS: Windows
- OS Version: Windows 10 Professionnel 22H2
- Node.js: 18.17.1
- Simple Scaffold Version: 1.8.0

#### Additional context

In the end, all I want is to be able to use a helper function in the template because I want to pass something like `some/folder/page` as `name` and be able to use the `page` part in my template, so I figured I would use a helper to extract this. But if there is another way, I'm opened to (and would prefer) not use the Node.js version and stick with the CLI.

Thanks!

**Comment by chenasraf** [comment]
Hi @opack, sorry for the trouble

I think there is a confusion (I may need to update the docs) - try importing the named Scaffold fn, instead of the default:

```ts
import { Scaffold } from 'simple-scaffold'
```

I tested with your setup and indeed it failed as you said, and for me this type of import worked.

If not only a problem with the docs, I might need to fix imports for `"type": "module"` packages.

As for your original problem, I think your way is one correct way but there might be simpler ways depending on your case:

1. One way as you said would be to add the helpers in the node setup:
  ```ts
  Scaffold({
    // ...
    helpers: { ... }
  })
  ```
  2. An alternative would be to only provide the config via file, like so:
    ```
    // file: scaffold.js (can be named anything)

    // `config` var holds the (so-far) parsed config
    export default (config) => ({
      some_key: {
        // ...
        data: {
          ...config,
          filename: path.basename(config.data.name), // `data.name` holds the given name to the scaffold
        }
      }
    })
   ```
  Then run the scaffold from CLI:
  ```shell
  npx simple-scaffold -c scaffold.js -k my_key /path/to/file
  ```
  3. Another way would be to pass it directly from the CLI like so:
  ```shell
  npx simple-scaffold  -d "{\"filename\": \"$(basepath /path/to/file)\"}"
  ```

**Comment by opack** [comment]
How come you are so great? 😬 The `{}` in the import did the trick, thanks!

Regarding my original problem, I used the second option, and ended up with this (for future readers):
```javascript
// scaffold.cjs
const path = require('node:path')

module.exports = (config) => ({
	route: {
		templates: ['tools/templates/page'],
		output: '.',
		data: {
			page: path.basename(config.name),
			route: config.name
		}
	}
})
```
Notes:
- It seems like there is no need to add the spread `...config` (which I think in your example should be at the same level as `data` and not inside it, if I'm correct).
- Being able to use this as an ESModule would be great also 😊 

It now works like a charm when I just run this from CLI:
```shell
$ npx simple-scaffold -c scaffold.cjs -k route "(main)/some/route/in/my/app"
Loading config from scaffold.cjs with key route
Loading config from file scaffold.cjs
Data: {
 "name": "(main)/some/route/in/my/app",
 "Name": "MainSomeRouteInMyApp",
 "page": "app",
 "route": "(main)/some/route/in/my/app"
}
Writing to C:\Users\micro\Projects\kazelio-kronos\src\routes\(main)\some\route\in\my\app\+page.svelte
Done.
Writing to C:\Users\micro\Projects\kazelio-kronos\src\routes\tests\(main)\some\route\in\my\app\+page.svelte
Done.
```
Now, I can use this to scaffold a route for my SvelteKit project and the corresponding testing page! 🥳 

Many many thanks for this awesome lib and your time and reactivity with the questions asked here 🙏

**Comment by chenasraf** [comment]
Glad the issue was resolved! And thank you very much for the kind words! Makes me happy to know other people make use of it.

Please let me know if there are any more issues. I will make time soon to work on improving ESM support and docs.

**Comment by opack** [comment]
I sure do use it, as I'm working with WebStorm and there is no "structure" scaffolding, only at file level. When I used VSCode I found an extension that did the job but was not perfect and a bit buggy. Your lib makes it possible to easily scaffold structures. Working with SvelteKit, I use it to scaffold the files necessary to create components, services, and now routes :-) For each one, I have 2 to 4 files to generate, and it allows me to save much time and make sure some basic things are always done, and done the same and right way 😉
